src/core/exec-invoke.c

```diff
@@ -3205,6 +3213,10 @@ static int apply_mount_namespace(
                 .temporary_filesystems = context->temporary_filesystems,
                 .n_temporary_filesystems = context->n_temporary_filesystems,
 
+                /* When DynamicUser=yes enforce that /tmp/ and /var/tmp/ are disconnected from the host's
+                 * directories, as they are world writable and ephemeral uid/gid will be used. */
+                .private_tmp_as_tmpfs = context->private_tmp == PRIVATE_TMP_DISCONNECTED,
```

Comment:
hmm, I think we shouldn't lose accuracy here. Please move the PrivateTmp typedef to namespace.h and pass the enum through as is.

THis is then more like we handle the various other enums such as ProtectHome, ProtectSystem, and so on.